src/object/authenticate.c

```diff
@@ -8717,6 +8718,49 @@ au_check_serial_authorization (MOP serial_object)
   return ret_val;
 }
 
+/*
+ * au_check_synonym_authorization () - check whether the current user is able to
+ *    modify synonym object or not.
+ * return: error code
+ *    ER_QPROC_CANNOT_UPDATE_SYNONYM
+ * synonym_object(in): synonym object pointer
+ */
+int
+au_check_synonym_authorization (MOP synonym_object)
+{
+  DB_VALUE creator_val;
+  MOP creator;
+  DB_SET *groups;
+  int ret_val;
+
+  CHECK_1ARG_ERROR (synonym_object);
+
+  ret_val = db_get (synonym_object, "owner", &creator_val);
+
+  if (ret_val != NO_ERROR || DB_IS_NULL (&creator_val))
+    {
```

Comment:
Call er_set inside the db_get function.
So, I will add only the code that checks the error code.

There seems to be no case where creator_val is NULL and NO_ERROR.

Reply:
It seems that there should be only the condition "ret_val != NO_ERROR".
However, other functions that start with "au_*" use the condition "ret_val == NO_ERROR && DB_IS_NULL(&creator_val)". 
This may need to be improved later.